ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/handlers/help/BbeHelpProvider.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.shell.cli.handlers.help;
+
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * Helper class to initialize functions related to help /TOPIC command.
+ *
+ */
+public class BbeHelpProvider {
+
+    private static final String BALLERINA_HOME =
+            System.getProperty("ballerina.home");
+    private static final String BBE_PATH = "/examples/";
+    private static final String DESCRIPTION = ".description";
+    private static final String INDEX_FILE = "index.json";
+
+    private final HashMap<String, String> helpMap;
+    private final List<String> listCommands;
+    private final HashMap<String, String> urlList;
+
+    public BbeHelpProvider() {
+        helpMap = new HashMap<>();
+        listCommands = new ArrayList<>();
+        urlList = new HashMap<>();
+        initHelpTopicCommand();
```

Comment:
A couple of questions.
- Do we really need to populate this each time an instance is created?
- Do we need to populate all the BBEs at once?

Reply:
Do we need to populate all the BBEs at once?

This was implemented to give code completions related to BBE topics.
First we populate a list with topic and call that list in `JlineSimpleCompleter`

This class only initiates two times
in the `JlineSimpleCompleter` and in the `HelpCommand` class

Reply:
Updated in 9bd04218bfdb8c681a91e0c7e1ba2ae46ecc3f0a